### PR TITLE
fix(types): rsdoctor types package change the dependency of @rspack/core

### PR DESCRIPTION
**Pull request by easy1090**
## Summary
fix(types): rsdoctor types package change the dependency of @rspack/core.

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/types/package.json [hunk 1]
```diff
@@ -31,6 +30,9 @@
     "tslib": "2.4.1",
     "typescript": "^5.2.2"
   },
+  "peerDependencies": {
+    "@rspack/core": "^0.5.1"
```
Comment (chenjiahan) [inline packages/types/package.json:34]: should we add optional meta

Comment (easy1090) [reply packages/types/package.json:34]: Fixed, please double-check.

Comment (easy1090) [reply packages/types/package.json:34]: > should we add optional meta

Fixed, please double-check.

Comment (easy1090) [reply packages/types/package.json:34]: > should we add optional meta

Fixed, please double-check.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**